docs/DeveloperGuide.md

```diff
@@ -1,38 +1,252 @@
+
+<!-- @@author yakultbottle -->
+
+### Storage
+
+**API**: Storage.java
+
+Entries are stored in Comma Separated Values(CSV) format. Fields read from left
+to right are: **Item type**, **Quantity**, and **Expiry date**(optional). Items
+may or may not have an expiry date, but all possess an Item type and a Quantity.
+
+Example of stored entries:
+
+```
+panadol,1
+panadol,30,2024-08-03
+panadol,20,2024-08-09
+bandage,34
+```
+
+The Storage class depends on self-defined classes PillException, Item, and
+ItemMap. While it has other dependencies, such as File and FileWriter from
+the Java standard library, PillException is the only custom class it depends on.
+
+<!-- @@author cnivedit -->
+
+### Logging
 
+**API**: PillLogger.java
+
+<img src = "diagrams/PillLogger.png" alt="Component Diagram for PillLogger"/>
+
+The project uses the `java.util.logging` package for logging, with PillLogger serving as a centralized utility class to
+handle logging across the entire application. PillLogger implements the singleton pattern by maintaining a single static
+logger instance, which manages log creation, configuration, and output redirection.
+
+#### Key Components
+- File Output Configuration: The log level for file output is set by the `fileHandler.setLevel()` call, using `Level.ALL` to
+capture all events during execution. The log file, named according to the `FILE_NAME` attribute, is created in the
+directory specified by `PATH`.
+
+- Console Output Configuration: Console output is managed by `consoleHandler.setLevel()`. To maintain a clean terminal
+output for end-users, console logging is set to `Level.OFF` by default, ensuring it is suppressed unless required for
+debugging.
+
+#### Resilience and Error Handling
+In the event of a failure in log file creation, PillLogger logs the error to the console and allows the application to
+continue running. This design ensures the application’s functionality is not hindered by logging setup issues.
+
+#### API Access
+PillLogger exposes a single public method, `getLogger()`, which provides application-wide access to the singleton Logger
+instance. Classes within the application use `getLogger()` to record events, without needing to set up or manage their own
+loggers.
+
+<!-- @@author yakultbottle -->
+
+### Item and ItemMap
+
+The Item class has three private variables, a name, a quantity, and an
+expiry date. An Item may or may not have an expiry date, so we store it
+as an Optional, which handles empty values for us without using null. 
+
+![](diagrams/Item-ClassDiagram.png)
+
+Quantity will always be a positive integer, and if no quantity is specified
+in the constructor, the default value is 1. Similarly, if no value is provided
+for expiry date, then it will be an Optional.empty().
+
+The ItemMap class contains a key-value pair, implemented as a Map, from the
+item name(String) to the item(TreeSet\<Item>)
+
+![](diagrams/ItemMap-ClassDiagram.png)
```

Comment:
I dont think this follows the standard also, I am not sure what the dotted box "item" mean on comparable

Reply:
should do comparable:item